wallet/src/utils/cryptography/mnemonics.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import bip39 from 'bip39-light';
+import nacl from 'tweetnacl';
+
+import type { Ed25519KeypairData } from '@mysten/sui.js';
+
+/**
+ * Generate a 12-word random mnemonic and keypair using crypto.randomBytes
+ * under the hood, defaults to 128-bits of entropy.
+ * @returns a tuple of mnemonic and keypair. The mnemonics is a 12-word string
+ * split by spaces.
+ */
+export function generateMnemonicsAndKeypair(): [string, Ed25519KeypairData] {
+    const mnemonics = bip39.generateMnemonic();
+    return [mnemonics, getKeypairFromMnemonics(mnemonics)];
+}
+
+/**
+ * Derive public key and private key from the Mnemonics
+ * @param mnemonics a 12-word seed phrase
+ * @returns public key and private key
+ */
+export function getKeypairFromMnemonics(mnemonics: string): Ed25519KeypairData {
```

Comment:
Typically BIP39 implementations expect an optional password (or passphrase), do we assume here that it's always empty?

See https://www.cs.utexas.edu/users/moore/acl2/manuals/current/manual/index-seo.php/BITCOIN____BIP39-MNEMONIC-TO-SEED

and https://github.com/maciejhirsz/tiny-bip39/blob/master/src/seed.rs#L33